subworkflows/local/shortread_adapterremoval.nf

```diff
@@ -0,0 +1,89 @@
+                                                .mix(
+                                                    ADAPTERREMOVAL_PAIRED.out.collapsed_truncated,
+                                                    ADAPTERREMOVAL_PAIRED.out.singles_truncated,
+                                                    ADAPTERREMOVAL_PAIRED.out.pair1_truncated,
+                                                    ADAPTERREMOVAL_PAIRED.out.pair2_truncated
+                                                    )
+                                                .map {
+                                                    meta, reads ->
+                                                        def meta_new = meta.clone()
+                                                        meta_new['single_end'] = true
+
+                                                        [ meta_new, reads ]
+                                                    }
+                                                    .groupTuple(by: 0)
+
+
+        ch_adapterremoval_reads_prepped_pe = CAT_FASTQ ( ch_adapterremoval_for_cat ).reads
+
+        ch_adapterremoval_reads_prepped = ch_adapterremoval_reads_prepped_pe.mix( ADAPTERREMOVAL_SINGLE.out.singles_truncated )
+
+    } else if ( params.shortread_clipmerge_mergepairs && params.shortread_clipmerge_excludeunmerged ) {
+            ch_adapterremoval_for_cat = ADAPTERREMOVAL_PAIRED.out.collapsed
+                                                    .mix( ADAPTERREMOVAL_PAIRED.out.collapsed_truncated )
+                                                    .map {
+                                                        meta, reads ->
+                                                            def meta_new = meta.clone()
+                                                            meta_new['single_end'] = true
+
+                                                            [ meta_new, reads ]
+                                                        }
+                                                        .groupTuple(by: 0)
+
+            ch_adapterremoval_reads_prepped_pe = CAT_FASTQ ( ch_adapterremoval_for_cat ).reads
+
+            ch_adapterremoval_reads_prepped = ch_adapterremoval_reads_prepped_pe.mix( ADAPTERREMOVAL_SINGLE.out.singles_truncated )
+
+    } else {
+
+        ch_adapterremoval_reads_prepped_pe = ADAPTERREMOVAL_PAIRED.out.pair1_truncated
+                                                .join( ADAPTERREMOVAL_PAIRED.out.pair2_truncated )
+                                                .groupTuple(by: 0)
+                                                .map { meta, pair1, pair2 ->
+                                                        [ meta, [ pair1, pair2 ].flatten() ]
+                                                }
+
+
+        ch_adapterremoval_reads_prepped = ch_adapterremoval_reads_prepped_pe
+                                    .mix( ADAPTERREMOVAL_SINGLE.out.singles_truncated )
```

Comment:
As before 

```suggestion
        ch_adapterremoval_reads_prepped = ADAPTERREMOVAL_PAIRED.out.pair1_truncated
                                                .join( ADAPTERREMOVAL_PAIRED.out.pair2_truncated )
                                                .groupTuple()
                                                .map { meta, pair1, pair2 ->
                                                        [ meta, [ pair1, pair2 ].flatten() ]
                                                }
                                    .mix( ADAPTERREMOVAL_SINGLE.out.singles_truncated )
```

Reply:
Also, I'm surprised this works. The output of `groupTuple` should be `[meta, [pair1, pair2]]`, no?

Reply:
That is a good point... it didn't work at first as it resulted in a double nested list for some reason. I'll test this later though to double check.

Reply:
Nope, 

```
        ch_adapterremoval_reads_prepped_pe = ADAPTERREMOVAL_PAIRED.out.pair1_truncated
                                                .join( ADAPTERREMOVAL_PAIRED.out.pair2_truncated )
                                                .groupTuple()
                                                .dump(tag: "hello")
```

results in 

```
[b6/499a12] Cached process > NFCORE_TAXPROFILER:TAXPROFILER:KRAKEN2_KRAKEN2 (2612)
[DUMP: hello] [['id':'2612', 'run_accession':'ERR5766176', 'instrument_platform':'ILLUMINA', 'single_end':false], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/3d/82855e3dff50213d0c3b9f8705e1fd/2612_ERR5766176.pair1.truncated.gz], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/3d/82855e3dff50213d0c3b9f8705e1fd/2612_ERR5766176.pair2.truncated.gz]]
[DUMP: hello] [['id':'2613', 'run_accession':'ERR5766181', 'instrument_platform':'ILLUMINA', 'single_end':false], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/f3/945c9e7b60b8e82abd10209856a497/2613_ERR5766181.pair1.truncated.gz], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/f3/945c9e7b60b8e82abd10209856a497/2613_ERR5766181.pair2.truncated.gz]]
WARN: Input tuple does not match input set cardinality declared by process `NFCORE_TAXPROFILER:TAXPROFILER:SHORTREAD_PREPROCESSING:FASTQC_PROCESSED` -- offending value: [[id:2612, run_accession:ERR5766176, instrument_platform:ILLUMINA, single_end:false], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/3d/82855e3dff50213d0c3b9f8705e1fd/2612_ERR5766176.pair1.truncated.gz], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/3d/82855e3dff50213d0c3b9f8705e1fd/2612_ERR5766176.pair2.truncated.gz]]
No such variable: Exception evaluating property 'tool' for nextflow.util.ArrayBag, Reason: groovy.lang.MissingPropertyException: No such property: tool for class: sun.nio.fs.UnixPath
Possible solutions: root
```

This is because pair1 / pair2 are separate channels I guess.

[.join()](https://www.nextflow.io/docs/latest/operator.html#join) simply puts everything inline